lib/tasks/ovirt.rake

```diff
@@ -0,0 +1,66 @@
+namespace :ovirt do
+  desc <<~END_DESC
+    Remove all Ovirt / RHEV compute resources & profiles from Foreman.
+    All hosts associated with these compute resources will be disassociated and kept.
+    All host groups with Ovirt compute resources & profiles will have their compute resource and compute profile removed.
+
+    Examples:
+      # foreman-rake ovirt:drop
+  END_DESC
+
+  task :drop => :environment do
+    compute_resources = ComputeResource.unscoped.where(type: 'Foreman::Model::Ovirt')
+
+    nothing_to_do if compute_resources.empty?
+
+    ask_user
+
+    puts 'Starting Ovirt integration drop ...'
+    compute_resources.each do |cr|
+      hosts = cr.hosts.unscoped
+      host_groups = cr.hostgroups.unscoped
+
+      hosts.each do |host|
+        puts "Disassociating host '#{host.name}' from the compute resource"
+        host.disassociate!
+      end
+
+      puts "Removing Compute Resource and Compute Profile from host groups."
+      host_groups.update_all(compute_resource_id: nil, compute_profile_id: nil)
+
+      puts "Removing compute resource '#{cr.name}'"
+      cr.destroy
+    end
+
+    the_end
+  end
+
+  def ask_user
+    puts "!!! WARNING !!! This operation is irreversible."
```

Comment:
Technically you can link them back up so I'm not sure it's really that irreversible. Just quite a bit of work.